app/components/autocomplete/at_mention/index.ts

```diff
@@ -42,9 +42,9 @@ const enhanced = withObservables([], ({database, channelId}: WithDatabaseArgs &
             switchMap((c) => of$(Boolean(c?.isGroupConstrained))),
         );
 
-        useChannelMentions = combineLatest([currentUser, currentChannel]).pipe(switchMap(([u, c]) => (u && c ? observePermissionForChannel(database, c, u, Permissions.USE_CHANNEL_MENTIONS, false) : of$(false))));
+        useChannelMentions = combineLatest([currentUser, currentChannel]).pipe(switchMap(([u, c]) => observePermissionForChannel(database, c, u, Permissions.USE_CHANNEL_MENTIONS, false)));
```

Comment:
This and other similar changes is just a bit of cleanup, by passing the check of `u` and `c` into the observe permission function.